README.md

```diff
@@ -33,108 +34,242 @@ Menu
 
-There are two reasons for this. The first is, of course: performance.
+For more information on how to get started running k6, please look at the [Running k6](https://docs.k6.io/docs/running-k6) documentation page. If you want to know more about making and measuring HTTP requests with k6, take a look [here](https://docs.k6.io/docs/http-requests) and [here](https://docs.k6.io/docs/k6http). And for information about the commercial Load Impact services like distributed cloud execution (the `k6 cloud` command) or Insights (`k6 run -o cloud`), you can visit [loadimpact.com](https://loadimpact.com/) or view the [Load Impact documentation](https://support.loadimpact.com/4.0/).
 
-If you read a file from disk on every single script iteration, it'd be needlessly slow; even if you cache the contents of the file and any imported modules, it'd mean the *first run* of the script would be much slower than all the others. Worse yet, if you have a script that imports or loads things based on things that can only be known at runtime, you'd get slow iterations thrown in every time you load something new.
+Overview
+--------
 
-But there's another, more interesting reason. By forcing all imports and file reads into the init context, we design for distributed execution. We know which files will be needed, so we distribute only those files. We know which modules will be imported, so we can bundle them up from the get-go. And, tying into the performance point above, the other nodes don't even need writable filesystems - everything can be kept in-memory.
+In this section we'll briefly explore some of the basic concepts and principles of how k6 works. If you want to learn more in-depth about the k6 scripting API, results output, and features, you can visit the full k6 documentation website at [https://docs.k6.io](docs.k6.io).
 
-As an added bonus, you can use this to reuse data between iterations (but only for the same VU):
+### Init and VU stages
 
-```js
-var counter = 0;
+Earlier, in the [Running k6](#running-k6) section, we mentioned that scripts must contain a `default` function. *"Why not just run my script normally, from top to bottom"*, you might ask - the answer is: we do, but code **inside** and **outside** your `default` function can do different things.
+
+Each virtual user (VU) executes your script in a completely separate JavaScript runtime, parallel to all of the other running VUs. Code inside the `default` function is called _VU code_, and is run over and over, for as long as the test is running. Code outside of the `default` function is called _init code_, and is run only once in each VU, when that VU is initialized.
```

Comment:
Each virtual user (VU) executes your script in a completely separate JavaScript runtime, parallel to all of the other running VUs. Code inside the `default` function is called _VU code_, and is run over and over, for as long as the test is running. Code outside of the `default` function is called _init code_, and is run only once per VU, when that VU is initialized.